airbyte-integrations/connectors/destination-gcs/build.gradle

```diff
@@ -1,51 +1,17 @@
 plugins {
-    id 'application'
     id 'airbyte-java-connector'
 }
 
 airbyteJavaConnector {
-    cdkVersionRequired = '0.2.0'
-    features = ['db-destinations']
-    useLocalCdk = false
+    cdkVersionRequired = '0.17.0'
+    features = ['db-destinations', 's3-destinations', 'gcs-destinations']
+    useLocalCdk = true
 }
 
-//remove once upgrading the CDK version to 0.4.x or later
-java {
-    compileJava {
-        options.compilerArgs.remove("-Werror")
-    }
-}
-
-airbyteJavaConnector.addCdkDependencies()
-
 application {
     mainClass = 'io.airbyte.integrations.destination.gcs.GcsDestination'
     applicationDefaultJvmArgs = ['-XX:+ExitOnOutOfMemoryError', '-XX:MaxRAMPercentage=75.0']
 }
 
 dependencies {
-
-    implementation platform('com.amazonaws:aws-java-sdk-bom:1.12.14')
-    implementation 'com.google.cloud.bigdataoss:gcs-connector:hadoop3-2.2.1'
-
-    // csv
-    implementation 'com.amazonaws:aws-java-sdk-s3:1.11.978'
-    implementation 'org.apache.commons:commons-csv:1.4'
-    implementation 'com.github.alexmojaki:s3-stream-upload:2.2.2'
-
-    // parquet
-    implementation ('org.apache.hadoop:hadoop-common:3.3.3') {
-        exclude group: 'org.slf4j', module: 'slf4j-log4j12'
-        exclude group: 'org.slf4j', module: 'slf4j-reload4j'
-    }
-    implementation ('org.apache.hadoop:hadoop-aws:3.3.3') { exclude group: 'org.slf4j', module: 'slf4j-log4j12'}
-    implementation ('org.apache.hadoop:hadoop-mapreduce-client-core:3.3.3') {
-        exclude group: 'org.slf4j', module: 'slf4j-log4j12'
-        exclude group: 'org.slf4j', module: 'slf4j-reload4j'
-    }
-    implementation ('org.apache.parquet:parquet-avro:1.12.0') { exclude group: 'org.slf4j', module: 'slf4j-log4j12'}
-    implementation ('com.github.airbytehq:json-avro-converter:1.1.0') { exclude group: 'ch.qos.logback', module: 'logback-classic'}
-
-    testImplementation 'org.apache.commons:commons-lang3:3.11'
-    testImplementation 'org.xerial.snappy:snappy-java:1.1.8.4'
```

Comment:
Most of the code in destination-gcs has been hoisted into the CDK's s3-destinations module, which is a much better home for it because the deps are the same.